wifiphisher/common/interfaces.py

```diff
@@ -465,6 +506,18 @@ def set_interface_mode(self, interface_name, mode):
         pyw.modeset(card, mode)
         pyw.up(card)
 
+    def add_card(self, interface):
```

Comment:
you can do it like this:
```python
def is_interface_valid(self, interface_name, mode=None):
    if mode != "internet" and self._is_iface_managed_by_nm:
            raiseInterfaceManagedByNetworkManagerError
    if mode == "monitor" and not interface_adapter.has_monitor_mode and
        not self._is_iface_managed_by_nm:
                raise InvalidInterfaceError(interface_name, mode)
            elif mode == "AP" and not interface_adapter.has_ap_mode:
    raise InvalidInterfaceError(interface_name, mode)

```

But perhaps it can be more simplified as I just wanted to outline the idea.